src/lib/ssh/exec.ts

```diff
@@ -6,15 +6,22 @@ import { getSSHConnectionForProject } from "./project.js";
 
 export type RunTarget = { appInstallationId: string } | { projectId: string };
 
+export type RunCommand =
+  | { command: string; args: string[] }
+  | { shell: string };
```

Comment:
Because I find this much more expressive[^1]: this way, it immediately becomes clear what the two different variants _mean_ -- whereas a plain `string` could mean anything, and would need to be documented thoroughly to make sense.

[^1]: https://refactoring.guru/smells/primitive-obsession